src/api/v1beta1/properties.go

```diff
@@ -177,14 +178,16 @@ func (dk *DynaKube) NeedsOneAgentPrivileged() bool {
 
 // ShouldAutoUpdateOneAgent returns true if the Operator should update OneAgent instances automatically.
 func (dk *DynaKube) ShouldAutoUpdateOneAgent() bool {
-	if dk.CloudNativeFullstackMode() {
+	switch {
```

Comment:
So I don't like switches either and prefer if's because if a switch has to be used due to multiple branches, it is usually a problem with the architecture. That is why I think this linter should still remain active. Whenever it complains about code, the offending code and the architecture surrounding it should be thought through again.